cleanlab_studio/studio/inference.py

```diff
@@ -45,13 +48,27 @@ def predict(
         Returns:
             predictions from batch as a numpy array, optionally also pandas dataframe of class probabilties
         """
-        csv_batch = self._convert_batch_to_csv(batch)
-        predictions, class_probabilities = self._predict_from_csv(csv_batch, timeout)
+
+        batchsize: int = BATCH_MAX_TABULAR if isinstance(batch, pd.DataFrame) else BATCH_MAX_TEXT
+        batched_preds: List[Predictions] = []
+        batched_probs: List[ClassProbablities] = []
+        num_batches = len(batch) // batchsize + (1 if len(batch) % batchsize != 0 else 0)
```

Comment:
nit: you could use `math.ceil` here
`num_batches = math.ceil(len(batch) / batchsize)`